consensus/src/dag/dag_driver.rs

```diff
@@ -224,10 +232,23 @@ impl DagDriver {
 
                         VerifiedEvent::CertifiedNodeRequestMsg(node_request) => {
                             self.handle_node_request(*node_request).await;
-                    }
-                    _ => unreachable!("reliable broadcast got wrong messsgae"),
+                        },
+                        _ => unreachable!("reliable broadcast got wrong messsgae"),
                     }
                 },
+
+                close_req = close_rx.select_next_some() => {
+                    if let Some(close_tx) = self.rb_close_tx.take() {
+                        let (ack_tx, ack_rx) = oneshot::channel();
+                        close_tx.send(ack_tx).expect("[DagDriver] failed to drop rb");
+                        ack_rx.await.expect("[DagDriver] failed to drop rb");
+                    }
+                    if let Ok(ack_sender) = close_req {
+                        ack_sender.send(()).expect("[DagDriver] Fail to ack shutdown");
+                    }
+                    break;
```

Comment:
synchronized wait for the ack from rb. Is that what we want? @sasha8

Reply:
Yes.